### PR TITLE
Feat/refactor semrel package

### PR DESCRIPTION
**Pull request by cliedeman**
Create Repository interface before creating Gitlab Repository

### CODE REVIEW

**Comment by cliedeman** [comment]
I split the work up into several commits. I still need to implement the Gitlab repository

File: pkg/semrel/github_test.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	"net/http/httptest"
+	"net/url"
+	"regexp"
+	"testing"
+
+	"github.com/Masterminds/semver"
+	"github.com/google/go-github/v30/github"
+)
+
+func TestNewGitlabRepository(t *testing.T) {
```
Comment (christophwitzko) [inline pkg/semrel/github_test.go:17]: I think this should be called `TestNewGithubRepository`

Comment (cliedeman) [reply pkg/semrel/github_test.go]: Whoops. Fixed

**Comment by christophwitzko** [comment]
Hi @cliedeman, thank you for working on this 🎉 
This pull request looks very good, can you rewrite the commit messages to match the [conventional commits format](https://www.conventionalcommits.org/en/v1.0.0/#summary). Then it will be added automatically to the release notes. 🙏

**Comment by christophwitzko** [comment]
Awesome, do you want to implement GitLab support next? I can add you to the go-semantic-release organization if you want so you can create a branch and work on it ☺️

**Comment by cliedeman** [comment]
Sure that would be good. I don't mind working in a fork for now but I expect this will take me a few days to implement and test

**Comment by christophwitzko** [comment]
That's awesome. Take your time :)

**Pull request merged**